src/main/java/bio/terra/workspace/service/resource/controlled/flight/create/notebook/CreateAiNotebookInstanceStep.java

```diff
@@ -0,0 +1,200 @@
+import bio.terra.cloudres.google.notebooks.InstanceName;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.generated.model.ApiGcpAiNotebookInstanceContainerImage;
+import bio.terra.workspace.generated.model.ApiGcpAiNotebookInstanceCreationParameters;
+import bio.terra.workspace.generated.model.ApiGcpAiNotebookInstanceVmImage;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.resource.controlled.ControlledAiNotebookInstanceResource;
+import bio.terra.workspace.service.workspace.WorkspaceService;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.services.notebooks.v1.model.ContainerImage;
+import com.google.api.services.notebooks.v1.model.Instance;
+import com.google.api.services.notebooks.v1.model.Operation;
+import com.google.api.services.notebooks.v1.model.VmImage;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import java.io.IOException;
+import java.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+
+/**
+ * A step for creating the AI Platform notebook instance in the Google cloud.
+ *
+ * <p>Undo deletes the created notebook instance.
+ */
+public class CreateAiNotebookInstanceStep implements Step {
+  private final Logger logger = LoggerFactory.getLogger(CreateAiNotebookInstanceStep.class);
+  private final CrlService crlService;
+  private final ControlledAiNotebookInstanceResource resource;
+  private final WorkspaceService workspaceService;
+
+  public CreateAiNotebookInstanceStep(
+      CrlService crlService,
+      ControlledAiNotebookInstanceResource resource,
+      WorkspaceService workspaceService) {
+    this.crlService = crlService;
+    this.resource = resource;
+    this.workspaceService = workspaceService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    String projectId = workspaceService.getRequiredGcpProject(resource.getWorkspaceId());
+    InstanceName instanceName = createInstanceName(projectId);
+    Instance instance = createInstanceModel(flightContext, projectId);
+
+    AIPlatformNotebooksCow notebooks = crlService.getAIPlatformNotebooksCow();
+    try {
+      OperationCow<Operation> creationOperation;
+      try {
+        creationOperation =
+            notebooks
+                .operations()
+                .operationCow(notebooks.instances().create(instanceName, instance).execute());
+      } catch (GoogleJsonResponseException e) {
+        // If the instance already exists, this step must have already run successfully. Otherwise
+        // retry.
+        if (HttpStatus.CONFLICT.value() == e.getStatusCode()) {
```

Comment:
AFAICT it's guaranteed to be on the initial create call. I can't find any documentation stating either way.
But if it can show up on the poll, we have an issue.

Reply:
I don't think so. I think that's code that we're very unlikely to need. If our create AI notebook flights are failing, we should debugging why that's happening, not trying to predict what the issue is.
It also seems like it's not the style for the GCP operations API to use the operations.get endpoint HTTP status code for the result of the logical operation. The `Operation` response has an `error` field for describing errors about the operation. 
Reminiscent of our JobReport HTTP status code conversations.
https://cloud.google.com/ai-platform/notebooks/docs/reference/rest/v1/projects.locations.operations
Here, we're throwing the RetryException with the error retrieved from the operation response body.